tgui/packages/tgui-say/handlers/incrementChannel.tsx

```diff
@@ -12,8 +12,15 @@ export const handleIncrementChannel = function (this: Modal) {
   if (radioPrefix === ':b ') {
     this.timers.channelDebounce({ mode: true });
   }
+  if (channel === 4) {
+    this.setState({
+      buttonContent: CHANNELS[channel],
+      channel: channel,
+    });
+    return;
+  }
   this.fields.radioPrefix = '';
-  if (channel === CHANNELS.length - 1) {
+  if (channel === CHANNELS.length - 2) {
```

Comment:
can we comment what this number is

Reply:
I kinda wanted to do this with minimal changes to the code. 1 was just cycling back to the start of the array. 2 is cycling back from the second to last element, so ignoring asay. I don't really know how to comment it better...

Reply:
that seems *really* fragile. if we add a new channel to the end then we're going to be skipping over that instead. i dont like this

Reply:
I can write an explanation why, and add a reminder in a comment in the definition file to keep the asay as the last one in array, is that okay?

Reply:
To be fair the entire proc should be stripped from magic numbers, since OOC is treated the same way. I might do that tomorrow, but for now I wanted to block the thing that multiple admins complained about already.

Reply:
> I can write an explanation why, and add a reminder in a comment in the definition file to keep the asay as the last one in array, is that okay?

comments aren't a substitute for good code :(

Reply:
Uhh, I changed it, can you look at it now?